dm/config/task.go

```diff
@@ -516,3 +529,21 @@ func (c *TaskConfig) SubTaskConfigs(sources map[string]DBConfig) ([]*SubTaskConf
 	}
 	return cfgs, nil
 }
+
+// checkDuplicateString checks whether the given string array has duplicate string item
+// if there is duplicate, it will return **all** the duplicate strings
+func checkDuplicateString(ruleNames []string) []string {
+	mp := make(map[string]bool, len(ruleNames))
+	dupeArray := make([]string, 0)
+	for _, name := range ruleNames {
+		if added, ok := mp[name]; ok {
+			if !added {
```

Comment:
seems that don't need judge `added`, if `ok` is true means this `ruleName` is duplicate

Reply:
I think its used to prevent `append` more than once.

Reply:
added is used to avoid to print "duplicate" duplicates.
For example, ["a", "a", "a"], "a" will be appened for two times.